### DISCUSSION TITLE
About locked wallet and wallet account changed

### DISCUSSION THREAD

**Discussion by 99sudarshan** [near/wallet-selector#620] [Q&A] [answered]
1. How to catch event and logout the user if wallet account switched in our code?
2.  After unlocked the wallet its still catch error "wallet is locked" unless refresh the page, how to solve it ?

**Comment by amirsaran3** [comment]
Hey @99sudarshan ,

1. https://github.com/near/wallet-selector/blob/main/packages/core/docs/api/selector.md#onevent-callback Here you can find more info about events. You can listen to the `accountsChanged` event.
2. Which wallet has this issue? Can you post more info about this. This maybe a problem with the wallet not wallet-selector.

**Reply by 99sudarshan** [reply]
**1. When account changed: Event triggering is not persistent for on('accountsChanged', ......)**

I have implemented like this when changing the account from wallet in my **walletSelectorContext.ts** but its didn't trigger on account changed even not rendered **walletSelectorContext.ts** whether its just logged from wallet selector maybe **store.ts** or **sender.ts**

```javascript useEffect(() => {
const subscription = selector?.on('accountsChanged', ({ walletId,accounts }) => {
console.log(account changed ${walletId})
})

return ()=>subscription?.remove()
}, [selector])

**Reply by 99sudarshan** [reply]
**2. When wallet is locked: Detect and request wallet unlock .**

I am facing this from sender wallet. when i unlocked the wallet its still catch error **"wallet is locked"** and I have implemented it like this and also dispatching some state statically to reload the page when user unlocked the wallet.

```javascript 
const handleConnectWallet: () => void = async () => {
try {
if (selector.isSignedIn()) {
const wallet = await selector.wallet()
// console.log("wallet",wallet)
// console.log("selector",selector)
const owner = await wallet.verifyOwner({
message: "Authorize this wallet to access your account"
})
if (owner) {
const response = await login(owner)
if (response?.status === 200) {
dispatch(changeSignState(true))
}
}
} else {
modal.show()
}
} catch (error) {
console.log(error)
dispatch(
toast.error({
message: t("unlock-wallet") || ""
})
)
dispatch(changeSignState(false))
setReload(true)
}
}

**Comment by amirsaran3** [comment, marked as answer]
Continued discussion here https://github.com/near/wallet-selector/issues/626
